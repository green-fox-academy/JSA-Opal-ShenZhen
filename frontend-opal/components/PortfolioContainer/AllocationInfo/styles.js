import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    allocationArea: {
        width: "92%",
        height: 280,
        marginTop: 10,

    },
    pie: {
        width: "100%",
        height: 220,

        backgroundColor: "#DFF1FD",
    },
    allocationButtons: {
        display: "flex",
        flexDirection: "row",
        justifyContent: "flex-start",

    },
    button: {
        height: 40,
        padding: 12,
        textAlign: "left",

        fontSize: 16
    },
})