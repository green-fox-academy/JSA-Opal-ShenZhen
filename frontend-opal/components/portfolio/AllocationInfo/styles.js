import { StyleSheet } from 'react-native';

let displayBorder = (width = 2, style = "solid", color = "black") => (
    {
        borderWidth: 0,
        borderStyle: style,
        borderColor: color,
    }
)

export default StyleSheet.create({
/* ------------------------ Allocation Area ------------------------ 
- a pie chart
- two buttons
------------------------------------------------------------------ */
    allocationArea: {
        // ...displayBorder(3, "solid", "#e17055"),
        width: "92%",
        height: 280,
        marginTop: 10,

    },
    pie: {
        // ...displayBorder(2, "solid", "black"),
        width: "100%",
        height: 220,

        backgroundColor: "#DFF1FD",
    },
    allocationButtons: {
        // ...displayBorder(2, "solid", "#fd79a8"),
        display: "flex",
        flexDirection: "row",
        justifyContent: "flex-start",

    },
    button: {
        // ...displayBorder(1, "solid", "#00cec9"),
        // width: 60,
        height: 40,
        padding: 12,
        textAlign: "left",

        fontSize: 16
    },
})