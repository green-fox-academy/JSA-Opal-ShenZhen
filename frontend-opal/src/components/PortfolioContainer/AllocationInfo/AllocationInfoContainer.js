import { connect } from 'react-redux';
import getTotalInvestment from 'components/common/getTotalInvestment';
import presenter from './AllocationInfoPresenter';
import groupBySectors from './groupBySectors';
import createPieData from './createPieData';

const mapStateToProps = ({ portfolio }) => {
  const totalInvestment = getTotalInvestment(portfolio);
  const sectors = groupBySectors(portfolio);

  Object.keys(sectors).forEach(sector => {
    sectors[sector].value =
      (
        sectors[sector].list.reduce((accumulativeValue, instrument) => {
          let result = accumulativeValue;
          result += instrument.amount * instrument.marketValue;
          return result;
        }, 0) / totalInvestment
      ).toFixed(2) * 100;
  });

  const legendData = createPieData.createLegendData(sectors);

  const pieData = createPieData.createPieData(sectors);

  const pieColor = createPieData.createPieColor(sectors);

  return { pieData, pieColor, legendData };
};

const mapDispatchToProps = dispatch => {
  return {
    fetchPortfolio: () => dispatch(thunk.fetchPortfolio())
  };
};

const getAllocationInfo = connect(mapStateToProps, mapDispatchToProps)(presenter);
export default getAllocationInfo;
