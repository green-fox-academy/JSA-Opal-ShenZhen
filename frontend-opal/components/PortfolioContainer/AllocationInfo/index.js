import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, TouchableOpacity } from 'react-native';
import { Card } from 'native-base';
import styles from './styles';

import PieChart from '../PieChart'

const AllocationInfo = ({pieData, pieColor, legendData}) => {
    return (
      <View style={styles.allocationArea}>
        <Text style={styles.title}>Allocation</Text>

        <Card>
          <View style={styles.pie}>
            <PieChart
              data={pieData}
              color={pieColor}
              legendData={legendData}
            />
          </View>

          <View style={styles.allocationButtons}>
            <TouchableOpacity onPress={() => console.log('button pressed')}>
              <Text style={styles.button}>SECTOR</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => console.log('button pressed')}>
              <Text style={styles.button}>GEOGRAPHY</Text>
            </TouchableOpacity>
          </View>

        </Card>
      </View>
    )
}

AllocationInfo.propTypes = {
  pieData: PropTypes.objectOf(PropTypes.object).isRequired,
  pieColor: PropTypes.arrayOf(PropTypes.array).isRequired,
  legendData: PropTypes.objectOf(PropTypes.object).isRequired
}


export default AllocationInfo;

