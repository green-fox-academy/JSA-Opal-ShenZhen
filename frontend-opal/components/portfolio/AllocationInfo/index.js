import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { Card } from 'native-base';
import styles from './styles';

import PieChart from '../PieChart'

const AllocationInfo = (props) => {
    return (
        <View style={styles.allocationArea}>
            <Text style={styles.title}>Allocation</Text>

            <Card>
                <View style={styles.pie}>
                    <PieChart
                        data={props.pieData}
                        color={props.pieColor}
                        legendData={props.legendData}
                    />
                </View>

                <View style={styles.allocationButtons}>
                    <TouchableOpacity onPress={() => console.log('button pressed')}>
                        <Text style={styles.button}>SECTOR</Text>
                    </TouchableOpacity>
                    <TouchableOpacity onPress={() => console.log('button pressed')}>
                        <Text style={styles.button}>GEOGRAPHY</Text>
                    </TouchableOpacity>
                </View>

            </Card>
        </View>
    )
}

export default AllocationInfo;

