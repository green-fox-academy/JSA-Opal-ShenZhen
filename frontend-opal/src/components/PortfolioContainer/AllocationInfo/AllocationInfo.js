import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { View, Text, TouchableOpacity } from 'react-native';
import { Card } from 'native-base';
import styles from './styles';

import PieChart from '../PieChart';

const AllocationInfo = ({ pieData, pieColor, legendData }) => (
  <View style={styles.allocationArea}>
    <Text style={styles.title}>Allocation</Text>

    <Card>
      <View style={styles.pie}>
        <PieChart data={pieData} color={pieColor} legendData={legendData} />
      </View>

      <View style={styles.allocationButtons}>
        <TouchableOpacity>
          <Text style={styles.button}>SECTOR</Text>
        </TouchableOpacity>
        <TouchableOpacity>
          <Text style={styles.button}>GEOGRAPHY</Text>
        </TouchableOpacity>
      </View>
    </Card>
  </View>
);

const mapStateToProps = state => ({
  pieData: state.portfolio.pieData,
  pieColor: state.portfolio.pieColor,
  legendData: state.portfolio.legendData
});

export default connect(mapStateToProps, null)(AllocationInfo);

AllocationInfo.defaultProps = {
  pieData: [],
  pieColor: [],
  legendData: []
};

AllocationInfo.propTypes = {
  pieData: PropTypes.arrayOf(
    PropTypes.shape({
      x: PropTypes.string,
      y: PropTypes.number,
      label: PropTypes.string
    })
  ),
  pieColor: PropTypes.arrayOf(PropTypes.string),
  legendData: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      symbol: PropTypes.objectOf(PropTypes.string)
    })
  )
};
