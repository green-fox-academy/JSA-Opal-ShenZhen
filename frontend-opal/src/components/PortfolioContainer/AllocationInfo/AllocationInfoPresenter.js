import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import { Card, CardItem, Button, Spinner } from 'native-base';
import styles from './styles';

import PieChart from '../PieChart';

const AllocationInfo = ({ pieData, pieColor, legendData }) => {
  return (
    <CardItem>
      <View style={styles.allocationArea}>
        <Card transparent>
          <Text style={styles.title}>Allocation</Text>
        </Card>

        <Card>
          <View style={styles.pie}>
            {!pieData[0] ? (
              <View
                style={{
                  height: '100%',
                  display: 'flex',
                  justifyContent: 'center',
                  alignItems: 'center'
                }}
              >
                <Spinner color="#e17055" />
              </View>
            ) : (
              <PieChart data={pieData} color={pieColor} legendData={legendData} />
            )}
          </View>

          <View style={styles.allocationButtons}>
            <Button style={styles.button}>
              <Text>SECTOR</Text>
            </Button>
            <Button style={styles.button}>
              <Text>GEOGRAPHY</Text>
            </Button>
          </View>
        </Card>
      </View>
    </CardItem>
  );
};

export default AllocationInfo;

AllocationInfo.defaultProps = {
  pieData: [],
  pieColor: [],
  legendData: []
};

AllocationInfo.propTypes = {
  pieData: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string,
      x: PropTypes.string,
      y: PropTypes.number
    })
  ),
  pieColor: PropTypes.arrayOf(PropTypes.string),
  legendData: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      symbol: PropTypes.objectOf(PropTypes.string)
    })
  )
};
