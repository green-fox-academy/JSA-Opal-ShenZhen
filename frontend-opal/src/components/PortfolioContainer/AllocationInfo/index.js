import AllocationInfo from './AllocationInfoContainer';

export default AllocationInfo;
