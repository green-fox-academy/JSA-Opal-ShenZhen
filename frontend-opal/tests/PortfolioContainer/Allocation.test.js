import React from 'react';
import renderer from 'react-test-renderer';

import AllocationInfo from '../../src/components/PortfolioContainer/AllocationInfo';

const pieData = [
  { x: 'IT', y: 40, label: '40%' },
  { x: 'Finance', y: 30, label: '30%' },
  { x: 'Energy', y: 30, label: '30%' }
];
const legendData = [
  { name: 'IT', symbol: { type: 'square' } },
  { name: 'Finance', symbol: { type: 'square' } },
  { name: 'Energy', symbol: { type: 'square' } }
];
const pieColor = ['#711702', '#A91600', '#E12C00'];

describe('<Header />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<AllocationInfo pieData={pieData} pieColor={pieColor} legendData={legendData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
