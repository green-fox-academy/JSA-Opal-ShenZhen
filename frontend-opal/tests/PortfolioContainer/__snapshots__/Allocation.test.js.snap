// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllocationInfo /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 2,
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomWidth": 0,
          "borderColor": "#ccc",
          "borderLeftWidth": 0,
          "borderRadius": 2,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "elevation": null,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      transparent={true}
    >
      <Text>
        Allocation
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderBottomWidth": 0.5,
          "borderColor": "#ccc",
          "borderLeftWidth": 0.5,
          "borderRadius": 2,
          "borderRightWidth": 0.5,
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexWrap": "nowrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#DFF1FD",
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#e17055"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "height": 80,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginBottom": 10,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 10,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text>
            SECTOR
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginBottom": 10,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 10,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text>
            GEOGRAPHY
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
